addons/mrp/report/mrp_report_bom_structure.py

```diff
@@ -58,17 +58,16 @@ def _compute_production_capacities(self, bom_qty, bom_data):
     @api.model
     def _get_report_values(self, docids, data=None):
         docs = []
-        for bom_id in docids:
-            bom = self.env['mrp.bom'].browse(bom_id)
+        for bom in self.env['mrp.bom'].browse(docids).exists():
```

Comment:
```suggestion
        for bom_id in docids:
            bom = self.env['mrp.bom'].browse(bom_id)
            if not bom:
                continue
```
`exists()` does more than you need here (like query the DB again), since we can just check if the recordset is empty instead.

Reply:
Done